controllers/managedfusionoffering_controller.go

```diff
@@ -85,11 +88,49 @@ func pluginReconcile(r *ManagedFusionOfferingReconciler) (ctrl.Result, error) {
 	if _, found := r.managedFusionOffering.Spec.Config["onboardingValidationKey"]; !found {
 		return ctrl.Result{}, fmt.Errorf("%s CR does not contain onboardingValidationKey entry", r.managedFusionOffering.Name)
 	}
+	if err := reconcileOnboardingValidationSecret(r); err != nil {
```

Comment:
Why not ?
```suggestion 
	if err := r.reconcileOnboardingValidationSecret(); err != nil {
```
This will introduce less changes when we copy the code from the old deployer

Reply:
I believe so. why do you think we will not?

Reply:
I was not able to make it work

Reply:
You can embed it, 
see here: https://go.dev/play/p/oY-ZBOU99JB

Reply:
Yes, this will work but in our case reconciler will be coming from another package so you cannot do the same thing.